src/dfx/src/commands/canister/call.rs

```diff
@@ -236,11 +242,30 @@ pub async fn exec(
             get_local_cid_and_candid_path(env, callee_canister, Some(canister_id))?
         }
     };
-    let maybe_candid_path = opts.candid.or(maybe_candid_path);
+    let method_type = if let Some(path) = opts.candid {
+        get_candid_type(CandidSource::File(&path), method_name)
+    } else if let Some(did) = fetch_remote_did_file(agent, canister_id).await {
+        get_candid_type(CandidSource::Text(&did), method_name)
+    } else if let Some(path) = maybe_local_candid_path {
+        eprintln!("DEPRECATION WARNING: Cannot fetch Candid interface from canister metadata, reading Candid interface from the local build artifact. In a future dfx release, we will only read candid interface from canister metadata.");
```

Comment:
Our other deprecation warnings go through `warn!`.  
```suggestion
        warn!(env.get_logger(), "DEPRECATION WARNING: Cannot fetch Candid interface from canister metadata, reading Candid interface from the local build artifact. In a future dfx release, we will only read candid interface from canister metadata.");
```